ai/src/main/java/org/triplea/ai/flowfield/odds/LanchesterDebugUiAction.java

```diff
@@ -0,0 +1,155 @@
+package org.triplea.ai.flowfield.odds;
+
+import static games.strategy.triplea.Constants.PLAYER_NAME_GERMANS;
+import static games.strategy.triplea.Constants.UNIT_ATTACHMENT_NAME;
+import static games.strategy.triplea.delegate.battle.BattleState.Side.DEFENSE;
+import static games.strategy.triplea.delegate.battle.BattleState.Side.OFFENSE;
+
+import games.strategy.engine.data.GameData;
+import games.strategy.engine.data.GamePlayer;
+import games.strategy.engine.data.Territory;
+import games.strategy.engine.data.Unit;
+import games.strategy.engine.data.UnitType;
+import games.strategy.triplea.attachments.UnitAttachment;
+import games.strategy.triplea.delegate.Matches;
+import games.strategy.triplea.delegate.power.calculator.CombatValueBuilder;
+import games.strategy.triplea.delegate.power.calculator.PowerStrengthAndRolls;
+import games.strategy.triplea.formatter.MyFormatter;
+import games.strategy.triplea.odds.calculator.AggregateResults;
+import games.strategy.triplea.odds.calculator.ConcurrentBattleCalculator;
+import games.strategy.triplea.ui.TripleAFrame;
+import java.awt.event.ActionEvent;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.Random;
+import java.util.function.Predicate;
+import java.util.stream.Collectors;
+import javax.swing.AbstractAction;
+import lombok.RequiredArgsConstructor;
+import org.triplea.ai.flowfield.FlowFieldAi;
+
+@RequiredArgsConstructor
+public class LanchesterDebugUiAction extends AbstractAction {
+  private static final long serialVersionUID = -919496373521710039L;
+
+  private final TripleAFrame frame;
+  private final FlowFieldAi ai;
+
+  @Override
+  public void actionPerformed(final ActionEvent e) {
```

Comment:
This debug item only works in Global40.  My hopes are to make this be able to calculate what the lanchenster attrition order is so that map makers could then add that to a file in their map.  It currently just runs a battle simulation and a lanchester calculation and displays the two results.

Reply:
> My hopes are to make this be able to calculate what the lanchenster attrition order is so that map makers could then add that to a file in their map. 

We need to discuss this. It's hell to support run-time dependencies that exist on hard disks. It is something we will forever have to support and potentially retrofit all maps to have such a file. A per-AI, per-map configuration IMO is a last resort, maybe a non-starter.

An example of this has been supporting save-game files, supporting network communication, and supporting variations of Map-XMLs, all have been challenges and each has resulted in bug reports and many hours of additional maintenance effort.